ckanext/dataset_subscriptions/actions/whatsup_notifications.py

```diff
@@ -29,3 +33,21 @@ def _process_dataset_subscriptions_data_dict_values(data_dict):
         if activity_streams_whatsup_notifications:
             del data_dict["activity_streams_whatsup_notifications"]
             data_dict["plugin_extras"]["dataset_subscriptions"]["activity_streams_whatsup_notifications"] = activity_streams_whatsup_notifications
+
+
+def send_whatsup_notification(activities, data_dict):
```

Comment:
this function sends only a single notification. The action should be sending notifications for the entire system (like with emails). Here you should filter out relevant activities and then send whatsapp one by one.

1. change the signature of this method to be exactly what you need: `send_whatsup_notification(activities, phonenumber):
2. create a separate method `send_whatsup_notificationS(context, data_dict)` which is going to aggregate recent activities for each user and then calling `send_whatsup_notification`for each one of them.